dist/debian/rules

```diff
@@ -91,6 +91,7 @@ endif
 		--mandir=/usr/share/man
 
 override_dh_auto_build:
+	mkdir -p /tmp/.cache/go-build ; export GOCACHE=/tmp/.cache/go-build
```

Comment:
We can't assume that /tmp is usable.  That should at least be `${TMPDIR:-/tmp`.

Note that GOROOT is already set globally to `$${TMPDIR:-/tmp}/appdebgo/go`.   How about setting GOCACHE to `$${TMPDIR:-/tmp}/appdebgo/cache`?  Does that work?

Reply:
@DrDaveD Unfortunately It doesn't work on PPA build environment.

I tried `$${TMPDIR:-/tmp}/appdebgo/cache. 
If I use `${TMPDIR:-/tmp}` for parent directory of `GOCACHE` directory, `GOCACHE` variable automatically set to `/sbuild-nonexistent/.cache/go-build` even if I explicitly make `${TMPDIR:-/tmp}/appdebgo/cache` directory.

I also tried `$${TMPDIR:-/tmp}/appdebgo/.cache/go-build`. It doesn't work as well.

On PPA build environment, It seems like we can not change build user and permissions like we do on GitHub build environment. 

On PPA build environment. `$HOME` is set to `/sbuild-nonexistent`.

```
HOME=/sbuild-nonexistent
```

I'll test other settings and will report here soon.

Reply:
@DrDaveD 

The following works.

```
ifdef TMPDIR
export GOCACHE=$(TMPDIR)/appdebgo/cache
else
export GOCACHE=/tmp/appdebgo/cache
endif
```

The following didn't work.

```
GOCACHE = $${TMPDIR:-/tmp}/appdebgo/cache
```

This set `GOCACHE` environment variable as `${TMPDIR:-/tmp}/appdebgo/cache` and got error `build cache is required, but could not be located: GOCACHE is not an absolute path`. This bash syntax is not evaluated when golang checks cache directory. golang uses the variable as is cause the above error.